### ISSUE TITLE
Deliver --overwrite_screenshots does not sync non-existent local screenshots

### ISSUE THREAD

**Issue by bennettrogers** [fastlane/fastlane#7613]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

I'm trying to use Deliver to update screenshots for multiple (iMessage) apps. I've tried using the `--overwrite_screenshots` flag, but either I'm not understanding what it's supposed to do, or it's working incorrectly.

My expectation is that if I set that flag, then once Deliver is done running the only screenshots associated with my app will be the ones that were in the `./fastlane/screenshots/iMessage` folder at runtime. But what I'm actually seeing is that Deliver *only* updates any screenshots that are in that folder that happen to have the exact same name as a pre-existing screenshot. This means that there is no way to e.g. completely remove all existing screenshots.

I believe that the issue is in the `deliver/upload_screenshots.rb` module. [Here](https://github.com/fastlane/fastlane/blob/master/deliver/lib/deliver/upload_screenshots.rb#L68) is where the screenshots are gathered, and [here](https://github.com/fastlane/fastlane/blob/master/deliver/lib/deliver/upload_screenshots.rb#L4-L23) is where it tries to remove them. The problem is the screenshots that are gathered to be removed are only the ones that happen to exist in the fastlane directory at that time - the same as the ones to be uploaded.

What I think needs to happen is something like what @jeremymlane does in [his script](https://github.com/fastlane/fastlane/issues/1529#issuecomment-250266237) in #1529. I would expect that all existing screenshots are completely removed and replaced with only those in the `./fastlane/screenshots/iMessage` directory at runtime.

Please let me know if I'm misunderstanding this flag, and if there's a better way to do what I need (and what @jeremymlane does with that script). If this is actually an issue, I'm happy to work on a pull request, but I would appreciate some guidance on how it should be exposed (A different flag? Use Spaceship to gather screenshots? Or maybe there's a helpful method somewhere else in Fastlane that I haven't seen that I can use to gather all existing screenshots?).

### Environment
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                 | Value                                       |
| ------------------- | ------------------------------------------- |
| OS                  | 10.12.2                                     |
| Ruby                | 2.3.3                                       |
| Bundler?            | false                                       |
| Git                 | git version 2.10.2                          |
| Installation Source | /usr/local/bin/fastlane                     |
| Host                | Mac OS X 10.12.2 (16C67)                    |
| Ruby Lib Dir        | /usr/local/Cellar/ruby/2.3.3/lib            |
| OpenSSL Version     | OpenSSL 1.0.2j  26 Sep 2016                 |
| Is contained        | false                                       |
| Xcode Path          | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version       | 8.2                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/fastlane/docs
# All available actions: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "1.111.0"

default_platform :ios

# custom settings
ENV["APP_NAME"]="XXX"

# general fastlane settings
ENV["FASTLANE_USER"]="XXX"
ENV["FASTLANE_TEAM_ID"]="XXX"

# match settings
ENV["MATCH_GIT_URL"]="XXX"
ENV["MATCH_APP_IDENTIFIER"]="XXX"

# gym settings
ENV["GYM_SCHEME"]=ENV["APP_NAME"]

################################

app_name = ENV["APP_NAME"]

platform :ios do
  xcodeproj = "./#{app_name}.xcodeproj"

  desc "Sync the certificates"
  lane :domatch do
    match(
        type: "appstore"
    )
  end

  desc "Build the project"
  lane :build do

    increment_build_number
    increment_version_number

    # export xcode schemes to a file so fastlane can use them
    recreate_schemes(
        project: xcodeproj,
    )

    # build the project
    gym
  end

  desc "Deploy a new version to the App Store"
  lane :deploy do

    # deploy to iTunes Connect
    deliver(
        force: true,
        price_tier: 1,
        app_rating_config_path: "./fastlane/ratings.json",
        overwrite_screenshots: true,
        app_icon: "./fastlane/metadata/app_icon.png",
        # primary_category: "",
        # secondary_category: "",
        # primary_first_sub_category: "",
        # primary_second_sub_category: "",
        # secondary_first_sub_category: "",
        # secondary_second_sub_category: "",
    )
  end

  error do |lane, exception|
    # slack(
    #   message: exception.message,
    #   success: false
    # )
  end
end


# More information about multiple platforms in fastlane: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Platforms.md
# All available actions: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md

# fastlane reports which actions are used
# No personal data is recorded. Learn more at https://github.com/fastlane/enhancer```
</details>

**No Appfile found**


### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.1.3   | ✅ Up-To-Date  |


### Loaded fastlane plugins:

**No plugins Loaded**


<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.0.0        |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.4        |
| claide                    | 1.0.1        |
| colored                   | 1.2          |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.4.2        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.4        |
| terminal-notifier         | 1.7.1        |
| unicode-display_width     | 1.1.2        |
| terminal-table            | 1.7.3        |
| plist                     | 3.2.0        |
| addressable               | 2.3.8        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| babosa                    | 1.0.2        |
| highline                  | 1.7.8        |
| commander                 | 4.4.1        |
| excon                     | 0.54.0       |
| faraday                   | 0.10.0       |
| unf_ext                   | 0.0.7.2      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20161129 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 1.9.0        |
| gh_inspector              | 1.0.2        |
| uber                      | 0.0.15       |
| representable             | 2.3.0        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| hurley                    | 0.2          |
| little-plugger            | 1.1.4        |
| multi_json                | 1.12.1       |
| logging                   | 2.1.0        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.9.20       |
| json                      | 1.8.3        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.0        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.1.1        |
| bundler                   | 1.13.6       |
| faraday_middleware        | 0.10.1       |
| i18n                      | 0.7.0        |
| thread_safe               | 0.3.5        |
| tzinfo                    | 1.2.2        |
| minitest                  | 5.8.5        |
| activesupport             | 4.2.7.1      |
| io-console                | 0.4.5        |
</details>


*generated on:* **2016-12-20**
</details>
